### ISSUE TITLE
Questions

### ISSUE THREAD

**Issue by kharmabum** [macmade/user-defaults#1]
(1) Every time a `Preferences` object is initialized the values in `NSUserDefaults` are set for each property's keypath, overriding any values that were set previously by the application.

(2) When the `Preferences` object is initialized the values of its properties are not set to the current values in `NSUserDefaults`.

Are (1) and (2) intended behavior?

**Comment by wdvr** [comment]
The code uses a file to read defaults: https://github.com/macmade/user-defaults/blob/42322fbaa7406633c09bb3d3bccf706224fc5902/swift/Preferences.swift#L39 

If you didn't add that resource, probably the `init` returns early.

Either you should add a plist resource, called Default.plist (can be empty), or remove line 39-44 (swift code).

**Comment by macmade** [comment]
Thank you for the comments guys. : )

@kharmabum is right, based on the blog article's code. I have just updated it, so it first read from the defaults before adding the observers.  
This was of course not the intended behaviour. The code here was right, but I tried to keep the article as simple as possible, hence the unfortunate omission.

As @wdvr said, I use a `Defaults.plist` file to get default values.  
This is not required, of course, but I find it easier this way.

@kharmabum if you take a look at the actual Swift code, you'll see:

    for c in Mirror( reflecting: self ).children
    {
        guard let key = c.label else
        {
            continue
        }
        
        self.setValue( UserDefaults.standard.object( forKey: key ), forKey: key )
        self.addObserver( self, forKeyPath: key, options: .new, context: nil )
    }

See the `self.setValue` call, before adding the observer?  
This will read from the user defaults, ensuring the instance is initialised with values from the suer defaults.
